schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java

```diff
@@ -0,0 +1,158 @@
+    private PaymentTypeEnum paymentType;
+    private Double fee;
+    private String feeReason;
+    private boolean enabled;
+    private String disabledReason;
+
+    private PaymentOption() {
+    }
+
+    private static String formatNumber(BigDecimal number, int precision) {
+        number = number.setScale(2, BigDecimal.ROUND_DOWN);
+        DecimalFormat df = new DecimalFormat();
+        df.setMaximumFractionDigits(precision);
+        df.setMinimumFractionDigits(precision);
+        df.setGroupingUsed(false);
+        return df.format(number);
+    }
+
+    public PaymentTypeEnum getPaymentType() {
+        return paymentType;
+    }
+
+    public void setPaymentType(PaymentTypeEnum paymentType) {
+        this.paymentType = paymentType;
+    }
+
+    public String getFeeReason() {
+        return feeReason;
+    }
+
+    public void setFeeReason(String feeReason) {
+        this.feeReason = feeReason;
+    }
+
+    public Double getFee() {
+        return fee;
+    }
+
+    public void setFee(Double fee) {
+        this.fee = fee;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public String getDisabledReason() {
+        return disabledReason;
+    }
+
+    public void setDisabledReason(String disabledReason) {
+        this.disabledReason = disabledReason;
+    }
+
+    public static class CashPaymentOption extends PaymentOption {
+        private static final String DisabledReason = "Not supported by supplier";
+
+        public CashPaymentOption() {
+            setPaymentType(PaymentTypeEnum.Cash);
+            setDisabledReason(DisabledReason);
+            setEnabled(false);
+            setFee(BigDecimal.ZERO.doubleValue());
+        }
+    }
+
+    public static class InvoicePaymentOption extends PaymentOption {
+        private static final Logger Log = LoggerFactory.getLogger(InvoicePaymentOption.class);
+        private static final long AllowedDays = 7;
+        private final static long InvoiceDaysBeforePaymentRequiredInMillis = 86400000 * AllowedDays;
+        private static final String DisabledReason = "For Invoice payments a minimum of " + AllowedDays + " days is required before the first activity";
+        private static final String FeeReason = createFeeReason();
+
+        public InvoicePaymentOption(List<ShoppingCart.Item> items) {
+            setPaymentType(PaymentTypeEnum.Invoice);
+            setDisabledReason(DisabledReason);
+            setEnabled(calculateEnabled(items));
+            setFee(InvoicePaymentProvider.INVOICE_FEE_FLAT.doubleValue());
+            setFeeReason(FeeReason);
+        }
+
+        private static String createFeeReason() {
+            String fee = formatNumber(InvoicePaymentProvider.INVOICE_FEE_FLAT, 2);
+            return "<nobr>For Electronic Invoice payments</nobr> there is a handling fee of <b><nobr>CHF " + fee + "</nobr></b>";
```

Comment:
We could get rid of the text and pass the numeric values over to front end and do the translation on front end.